src/endpoint/s3/ops/s3_get_object.js

```diff
@@ -15,14 +15,21 @@ async function get_object(req, res) {
     const noobaa_trigger_agent = agent_header && agent_header.includes('exec-env/NOOBAA_FUNCTION');
     const encryption = s3_utils.parse_encryption(req);
 
-    const object_md = await req.object_sdk.read_object_md({
+    let get_from_cache = false;
+    if (req.query.get_from_cache === 'true') {
+        get_from_cache = true;
+    }
+    let md_params = {
         bucket: req.params.bucket,
         key: req.params.key,
         version_id: req.query.versionId,
         md_conditions: http_utils.get_md_conditions(req),
         encryption
-    });
-
+    };
+    if (get_from_cache) {
+        md_params.get_from_cache = true;
+    }
```

Comment:
```suggestion
    if (req.query.get_from_cache !== undefined) {
        params.get_from_cache = true;
    }
```

You don't need to go through another local variable `get_from_cache`, and you don't need to require `req.query.get_from_cache` to have a specific literal string value of `"true"` because it is easy to forget and use the simple form of `https://endpoint?get_from_cache` which actually gets parsed to an empty string:

```
$ node -p 'querystring.parse("get_from_cache&x=true&x=true")'
[Object: null prototype] { get_from_cache: '', x: [ 'true', 'true' ] }
```

So checking that `query.key` is not undefined means all the url query options work, even accidental var repetition that will result in an array.

Reply:
ok